### PR TITLE
IGNITE-14058: Bool arrays now return as bool array, not int array

### PR DESCRIPTION
**Pull request by isapego**

### CODE REVIEW

File: pyignite/datatypes/primitive_arrays.py [hunk 1]
```diff
@@ -297,3 +297,10 @@ class BoolArrayObject(PrimitiveArrayObject):
     _type_id = TYPE_BOOLEAN_ARR
     primitive_type = Bool
     type_code = TC_BOOL_ARRAY
+
+    @classmethod
+    def to_python(cls, ctype_object, *args, **kwargs):
+        result = []
```
Comment (ivandasch) [inline pyignite/datatypes/primitive_arrays.py:303]:
```
if not ctype_object:
     return None
result = False * ctype_object.length
for i in range(ctype_objet.length):
     result[i] = ctype_object.data[i] != 0
```

Comment (ivandasch) [reply pyignite/datatypes/primitive_arrays.py:303]:
```
>>> def append_creation(n):
...     res = []
...     for _ in range(n):
...         res.append(False)
...     return res
... 
>>> timeit("False * 100", number=10000)
0.0005339730014384259
>>> timeit(lambda: append_creation(100), number=10000)
0.07856238499880419
>>> 
```

Comment (isapego) [reply pyignite/datatypes/primitive_arrays.py:303]: So that's how you do `.reserve()` in python :)

Comment (ivandasch) [reply pyignite/datatypes/primitive_arrays.py]: Sorry for mistake

```
if not ctype_object:
     return None
result = [False] * ctype_object.length
for i in range(ctype_objet.length):
     result[i] = ctype_object.data[i] != 0
```

>>> timeit("[False] * 100", number=10000)
0.014093294998019701
>>> timeit(lambda: append_creation(100), number=10000)
0.07997436200093944
>>>

**Review by ivandasch — APPROVED** [review]

**Pull request closed without merging**